homeassistant/components/fritz/image.py

```diff
@@ -0,0 +1,80 @@
+"""FRITZ image integration."""
+
+from __future__ import annotations
+
+from io import BytesIO
+import logging
+
+from homeassistant.components.image import ImageEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import EntityCategory
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.util import dt as dt_util, slugify
+
+from .common import AvmWrapper, FritzBoxBaseEntity
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up guest WiFi QR code for device."""
+    avm_wrapper: AvmWrapper = hass.data[DOMAIN][entry.entry_id]
+
+    guest_wifi_info = await hass.async_add_executor_job(
+        avm_wrapper.fritz_guest_wifi.get_info
+    )
+
+    if not guest_wifi_info.get("NewEnable"):
+        return
+
+    async_add_entities(
+        [FritzGuestWifiQRImage(avm_wrapper, entry.title, guest_wifi_info["NewSSID"])]
+    )
+
+
+class FritzGuestWifiQRImage(FritzBoxBaseEntity, ImageEntity):
+    """Implementation of the FritzBox guest wifi QR code image entity."""
+
+    _attr_content_type = "image/png"
+    _attr_entity_category = EntityCategory.DIAGNOSTIC
+
+    def __init__(
+        self, avm_wrapper: AvmWrapper, device_friendly_name: str, ssid: str
+    ) -> None:
+        """Initialize the image entity."""
+        self._attr_name = f"{device_friendly_name} {ssid} QR-Code"
```

Comment:
Instead of adding device friendly name, use `_attr_has_entity_name = True` and set name to just `ssid`. I don't think that adding "QR-Code" to the name is necessary, that's clear from the picture.

Reply:
we already have switches (_to turn on/off the wifi_) which are named like `f"{device_friendly_name} {ssid}`, so this might confuse the user?